### PR TITLE
 able to fetch filter timeline tweets  working on testCases

### PR DESCRIPTION
**Pull request by raj8789**
able to fetch filtered that timeline which contains the tweets

### CODE REVIEW

File: src/main/java/com/resource/TwitterController.java [hunk 1]
```diff
@@ -68,4 +68,23 @@ public Response getTweets() {
         }
         return Response.ok(tweets).build();
     }
+
+    @GET
+    @Path("/filter")
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response getFilterTweets(@QueryParam("searchKey") String searchKey)
+    {
+        System.out.println();
```
Comment (anasuya1117) [inline src/main/java/com/resource/TwitterController.java]: remove `System.out.println`

File: src/main/java/com/service/TwitterImpl.java [hunk 2]
```diff
@@ -25,6 +24,8 @@
 
 
     TwitterResponse twitterResponse;
+    //private TwitterResponse;
```
Comment (anasuya1117) [inline src/main/java/com/service/TwitterImpl.java]: remove all unused lines

File: src/main/java/com/service/TwitterImpl.java [hunk 3]
```diff
@@ -87,4 +88,40 @@ public Status sendTweets(String tweet) {
         }
         return twitList;
     }
+    public List<TwitterResponse> getTweetBasedOnMyFilter(String tweet)
+    {
+        String twitterHandle;
+        String name;
+        String message;
+        String profileImageUrl;
+        Date created =null;
```
Comment (anasuya1117) [inline src/main/java/com/service/TwitterImpl.java]: Declare the variables where u are using.

**Review by anasuya1117 — APPROVED** [review]
Address the comments and merge the PR

File: src/main/java/com/service/TwitterImpl.java [hunk 4]
```diff
@@ -87,4 +88,40 @@ public Status sendTweets(String tweet) {
...
+                Status s = statuses.get(i);
+                profileImageUrl = s.getUser().getProfileImageURL();
+                name = s.getUser().getName();
+                twitterHandle =s.getUser().getScreenName();
+                message = s.getText();
+                created = s.getCreatedAt();
+                Format dateFormat = new SimpleDateFormat("dd-MM-yyyy HH:mm:ss");
+                String date = dateFormat.format(created);
+                twitterResponse= new TwitterResponse(message,twitterHandle,name,profileImageUrl,date);
+                twitList.add(twitterResponse);
```
Comment (SupriyaPamulapati) [inline src/main/java/com/service/TwitterImpl.java]: i feel u can call the get tweets here and filter it , no need to write the same code again

File: src/main/java/com/service/TwitterImpl.java [hunk 5]
```diff
@@ -87,4 +88,40 @@ public Status sendTweets(String tweet) {
...
+        } catch (TwitterException e) {
+            logger.error("Error Occur", e);
+            throw new InternalServerErrorException("Server error, could not fetch tweet");
+        }
+        if (twitList.isEmpty()) {
+            logger.info("You Have No Tweets On your Timeline");
+        }
+        int end=tweet.length();
+        CharSequence charSequence=tweet.subSequence(0,end);
+        filterTwitList=twitList.stream().filter(t->t.getMessage().contains(charSequence)).collect(Collectors.toList());
```
Comment (SupriyaPamulapati) [inline src/main/java/com/service/TwitterImpl.java:89]: u can declare the variable here  only

File: src/test/com/TwitterImplTest.java [hunk 6]
```diff
@@ -9,49 +10,88 @@
...
+        when(s1.getText()).thenReturn(message);
+        when(s1.getCreatedAt()).thenReturn(created);
+        when(twitter.getHomeTimeline()).thenReturn(responseList);
+        twitListExpected.add(null);
+        List<TwitterResponse> actualListExpected = twitterImpl.getTweetBasedOnMyFilter("oo");
+        Assert.assertEquals(twitListExpected,actualListExpected);
+    }
     @Test
     public void testCase_fetchNoTweetOnTimeline_successCase() throws TwitterException {
         ResponseList<Status> responseList = mock(ResponseList.class);
```
Comment (SupriyaPamulapati) [inline src/test/com/TwitterImplTest.java:93]: u can add some more test cases on filtering tweets like checking  case sensitive

File: src/main/java/com/resource/TwitterController.java (hunk 1 above)
Comment (raj8789) [reply src/main/java/com/resource/TwitterController.java]: removed

File: src/main/java/com/service/TwitterImpl.java (hunk 2 above)
Comment (raj8789) [reply src/main/java/com/service/TwitterImpl.java]: removed

File: src/main/java/com/service/TwitterImpl.java (hunk 3 above)
Comment (raj8789) [reply src/main/java/com/service/TwitterImpl.java]: declared where i am using

File: src/test/com/TwitterImplTest.java (hunk 6 above)
Comment (raj8789) [reply src/test/com/TwitterImplTest.java:93]: String we Pass from postman is already case sensitive it while fetching tweet while filtering will treat god & GOD both different

File: src/main/java/com/service/TwitterImpl.java (hunk 4 above)
Comment (raj8789) [reply src/main/java/com/service/TwitterImpl.java]: done with this part

**Pull request merged**